configargparse.py

```diff
@@ -399,11 +400,16 @@ def parse_known_args(self, args = None, namespace = None,
 
         # normalize args by converting args like --key=value to --key value
         normalized_args = []
+        explicit_param_value_mapping = []
         for arg in args:
             if arg and arg[0] in self.prefix_chars and '=' in arg:
-                key, value =  arg.split('=', 1)
+                key, value = arg.split('=', 1)
                 normalized_args.append(key)
                 normalized_args.append(value)
+                # Remember that these two objects form a key=value pair, will be useful later in
+                # this function. We store the ids of the object, so that any comparison matches
+                # these precise strings, not any other string that would have the same content.
+                explicit_param_value_mapping.append((id(key), id(value)))
```

Comment:
I haven't tested this, but would it work to add quotes around the value (something like `if value.startswith("-") and not value.startswith('"'): value = '"' + value + '"'`) instead of creating this new `explicit_param_value_mapping` list and then rejoining the args?